src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -0,0 +1,145 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+
+@SuppressWarnings({"unused", "ConstantConditions"})
+class Exercise2 {
+
+    private static class MapHelper<T> {
+
+        private final List<T> source;
+
+        private MapHelper(List<T> source) {
+            this.source = source;
+        }
+
+        public static <T> MapHelper<T> from(List<T> source) {
+            return new MapHelper<>(source);
+        }
+
+        public List<T> getMapped() {
+            return new ArrayList<>(source);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         *
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+            List<R> result = new ArrayList<>();
+            source.forEach(item -> result.add(mapping.apply(item)));
+            return from(result);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         *
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            List<R> result = new ArrayList<>();
+            for (T t : source) {
+                result.addAll(flatMapping.apply(t));
+            }
```

Comment:
You can use `source.forEach` istead of loop

Reply:
Fixed